src/pulselistener/pulselistener/mercurial.py

```diff
@@ -115,12 +115,30 @@ def clean(self):
         base, patches = self.phabricator_api.load_patches_stack(self.repo, diff, default_revision='central')
         assert len(patches) > 0, 'No patches to apply'
 
+        # Load all the diffs details with commits messages
+        diffs = self.phabricator_api.search_diffs(
+            diff_phid=[p[0] for p in patches],
+            attachments={
+                'commits': True,
+            }
+        )
+        commits = {
+            diff['phid']: diff['attachments']['commits'].get('commits', [])
+            for diff in diffs
+        }
+
         # Apply the patches and commit them one by one
         for diff_phid, patch in patches:
-            logger.info('Applying patch', phid=diff_phid)
+            commit = commits.get(diff_phid)
```

Comment:
There are cases in which commits doesn't have a diff_phid?

Reply:
Someone could post a patch on phabricator without using a commit.